sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -123,8 +123,12 @@ class QueryExecution(val sparkSession: SparkSession, val logical: LogicalPlan) {
             .mkString("\t")
       }
     // SHOW TABLES in Hive only output table names, while ours outputs database, table name, isTemp.
-    case command: ExecutedCommandExec if command.cmd.isInstanceOf[ShowTablesCommand] =>
-      command.executeCollect().map(_.getString(1))
+    case command@ ExecutedCommandExec(showTables: ShowTablesCommand) =>
+      if (showTables.isExtended) {
+        command.executeCollect().map(_.getString(3))
```

Comment:
If we do not have it, any test case failed? 

This function is just for testing. If we do not have the related Hive output comparison. We can simplify it to 
```Scala
    case command @ ExecutedCommandExec(showTables: ShowTablesCommand) if !showTables.isExtended =>
      command.executeCollect().map(_.getString(1))
```